wgengine/magicsock/endpoint.go

```diff
@@ -914,9 +925,16 @@ func (de *endpoint) handlePongConnLocked(m *disco.Pong, di *discoInfo, src netip
 		}))
 	}
 
+	// XXX Previously this accepted any pong as a response to a
+	// CLI ping, including a rando heartbeat pong, but I added a
+	// tx ID and refactored the code a bit to get it set before we
+	// send the CLI ping.
 	for _, pp := range de.pendingCLIPings {
-		de.c.populateCLIPingResponseLocked(pp.res, latency, sp.to)
-		go pp.cb(pp.res)
+		if m.TxID == pp.tx {
+			de.c.logf("\n\n\nGOT CLI PONG sent tx %v received tx %v\n\n\n", pp.tx, m.TxID)
+			de.c.populateCLIPingResponseLocked(pp.res, latency, sp.to)
+			go pp.cb(pp.res)
+		}
 	}
 	de.pendingCLIPings = nil
```

Comment:
Now when you're not fully draining `pendingCLIPings` perhaps you have to keep the unprocessed items around for the next time?